### ISSUE TITLE
Support user-defined functions as options to abort a simulation

### ISSUE THREAD

**Issue by alahiff** [simvue-io/python-api#420]
For some codes there is a special way to cleanly stop a simulation, e.g. for FDS you create a `<CHID>.stop` file. I think JETTO has something similar. We should provide a way for users to define a function which does whatever is necessary which is executed when the run should be aborted.

This is something which a fire safety consultancy explicitly asked about.

**Comment by alahiff** [comment]
https://cloudhpc.cloud/2023/08/29/fds-soft-vs-hard-stop/

**Comment by kzscisoft** [comment]
Possibly solved by #500 

Suggest trying the following with this branch:

```python
def abort_callback() -> None:
    # Function to create soft abort file
    pass

with simvue.Run(abort_callback=abort_callback) as run:
    # Disable Simvue itself terminating the process
    run.config(abort_on_alert=False)
    # Add the FDS simulation process
    run.add_process(...)
```
